File: simulation/spiral-matrix-ii.cpp
class Solution {
public:
    vector<vector<int>> generateMatrix(int n) {
        int startx = 0;
        int starty = 0;
        int offset = 1;
        int loop = n/2;
        int count = 1;

        vector<vector<int>> res(n, vector<int>(n,0));

        while (loop){
            int i = startx;
            int j = starty;

            for (;j<n-offset;++j){
                res[i][j] = count++;
            }
            for (;i<n-offset;++i){
                res[i][j] = count++;
            }
            for (;j>startx; --j){
                res[i][j] = count++;
            }
            for (;i>startx; --i){
                res[i][j] = count++;
            }
            startx++;
            starty++;
            offset++;
            loop--;
        }

        if (n%2==1){
            int mid = n/2;
            res[mid][mid] = count++;
        }
        return res;
    }
};

File: matrix/spiral-matrix-ii.py
class Solution(object):
    def generateMatrix(self, n):
        """
        :type n: int
        :rtype: List[List[int]]
        """

        startx = 0
        starty = 0
        offset = 1
        loop = n//2
        result = [[0 for _ in range(n)] for _ in range(n)]
        count = 1

        while loop:
            i = startx
            j = starty

            while j < n-offset:
                result[i][j] = count
                count+=1
                j+=1
            while i < n-offset:
                result[i][j] = count
                count+=1
                i+=1
            while j > startx:
                result[i][j] = count
                count+=1
                j-=1
            while i > startx:
                result[i][j] = count
                count+=1
                i-=1
            startx+=1
            starty+=1
            offset+=1
            loop-=1
        
        if n%2==1:
            mid = n//2
            result[mid][mid] = count
        
        return result



        